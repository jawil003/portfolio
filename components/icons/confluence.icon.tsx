interface Props {
  fontSize: "large" | "normal" | "small";
}
export const ConfluenceIcon: React.FC<Props> = (
  props: Props
) => {
  return (
    <svg
      style={{
        fontSize:
          props.fontSize === "small"
            ? "0.5rem"
            : props.fontSize === "normal"
            ? "1rem"
            : "3.5rem",
        ...{
          fillRule: "evenodd",
          clipRule: "evenodd",
          strokeLinejoin: "round",
          strokeMiterlimit: 2,
        },
      }}
      width="1em"
      height="1em"
      viewBox="0 0 65 64"
      version="1.1"
    >
      <g id="Layer_2">
        <g id="Blue">
          <path
            d="M2.23,49.53C1.58,50.59 0.85,51.82 0.23,52.8C-0.321,53.732 -0.021,54.951 0.9,55.52L13.9,63.52C14.216,63.715 14.58,63.818 14.951,63.818C15.655,63.818 16.31,63.446 16.67,62.84C17.19,61.97 17.86,60.84 18.59,59.63C23.74,51.13 28.92,52.17 38.26,56.63L51.15,62.76C51.419,62.888 51.712,62.954 52.01,62.954C52.801,62.954 53.521,62.484 53.84,61.76L60.03,47.76C60.465,46.767 60.016,45.591 59.03,45.14C56.31,43.86 50.9,41.31 46.03,38.96C28.51,30.45 13.62,31 2.23,49.53Z"
            style={{
              fill: "url(#_Linear1)",
              fillRule: "nonzero",
            }}
          />
          <path
            d="M60.52,17.76C61.17,16.7 61.9,15.47 62.52,14.49C63.071,13.558 62.771,12.339 61.85,11.77L48.85,3.77C48.52,3.548 48.131,3.429 47.733,3.429C47.019,3.429 46.357,3.811 46,4.43C45.48,5.3 44.81,6.43 44.08,7.64C38.93,16.14 33.75,15.1 24.41,10.64L11.56,4.54C11.291,4.412 10.998,4.346 10.7,4.346C9.909,4.346 9.189,4.816 8.87,5.54L2.68,19.54C2.245,20.533 2.694,21.709 3.68,22.16C6.4,23.44 11.81,25.99 16.68,28.34C34.24,36.84 49.13,36.27 60.52,17.76Z"
            style={{
              fill: "url(#_Linear2)",
              fillRule: "nonzero",
            }}
          />
        </g>
      </g>
      <defs>
        <linearGradient
          id="_Linear1"
          x1="0"
          y1="0"
          x2="1"
          y2="0"
          gradientUnits="userSpaceOnUse"
          gradientTransform="matrix(-39.33,-22.6,22.6,-39.33,59.68,67.65)"
        >
          <stop
            offset="0"
            style={{
              stopColor: "rgb(0,82,204)",
              stopOpacity: 1,
            }}
          />
          <stop
            offset="0.18"
            style={{
              stopColor: "rgb(0,82,204)",
              stopOpacity: 1,
            }}
          />
          <stop
            offset="1"
            style={{
              stopColor: "rgb(38,132,255)",
              stopOpacity: 1,
            }}
          />
        </linearGradient>
        <linearGradient
          id="_Linear2"
          x1="0"
          y1="0"
          x2="1"
          y2="0"
          gradientUnits="userSpaceOnUse"
          gradientTransform="matrix(2371.5,682.635,-682.635,2371.5,187.58,-7.74211)"
        >
          <stop
            offset="0"
            style={{
              stopColor: "rgb(0,82,204)",
              stopOpacity: 1,
            }}
          />
          <stop
            offset="0.18"
            style={{
              stopColor: "rgb(0,82,204)",
              stopOpacity: 1,
            }}
          />
          <stop
            offset="1"
            style={{
              stopColor: "rgb(38,132,255)",
              stopOpacity: 1,
            }}
          />
        </linearGradient>
      </defs>
    </svg>
  );
};
