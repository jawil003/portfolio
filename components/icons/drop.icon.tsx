import React from "react";

interface Props {
  width?: string;
  height?: string;
  color?: string;
}

/**
 * An DropIcon React Component.
 * @author Jannik Will
 * @version 0.1
 */
const DropIcon: React.FC<Props> = ({
  width,
  height,
}) => {
  return (
    <svg
      viewBox="0 0 351 512"
      xmlns="http://www.w3.org/2000/svg"
      width={width}
      height={height}
    >
      <path
        d="M325.156 212.773c-13.715-34.277-32.781-69.832-56.676-105.675-40.527-60.79-80.527-101.024-82.207-102.703A14.989 14.989 0 00175.664 0a14.986 14.986 0 00-10.605 4.395c-1.68 1.683-41.68 41.914-82.207 102.703C58.957 142.94 39.89 178.496 26.18 212.773 8.809 256.2 0 297.77 0 336.336 0 433.196 78.805 512 175.664 512c96.863 0 175.668-78.805 175.668-175.668 0-38.562-8.809-80.133-26.176-123.559zM175.664 431.664c-52.562 0-95.332-42.766-95.332-95.332 0-8.281 6.719-15 15-15 8.285 0 15 6.719 15 15 0 36.023 29.309 65.332 65.332 65.332 8.285 0 15 6.719 15 15 0 8.285-6.715 15-15 15zm0 0"
        fill="#fff"
        xmlns="http://www.w3.org/2000/svg"
        data-original="#000000"
      />
    </svg>
  );
};

export default DropIcon;
