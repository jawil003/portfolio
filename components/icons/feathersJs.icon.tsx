export const FeathersJsIcon: React.FC = () => {
  return (
    <img
      width="56px"
      height="56px"
      src="pictures/feathersjs.jpg"
    />
  );
};
