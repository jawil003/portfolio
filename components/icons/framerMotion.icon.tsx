interface Props {
  fontSize: "large" | "normal" | "small";
}
export const FramerMotionIcon: React.FC<Props> = ({
  fontSize,
}) => {
  return (
    <svg
      width="1em"
      height="1em"
      style={{
        fontSize:
          fontSize === "small"
            ? "0.5rem"
            : fontSize === "normal"
            ? "1rem"
            : "3.5rem",
      }}
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fill="black"
        d="M0 14V1l6.5 6.5L13 1v13l-3.25-3.25L6.5 14l-3.25-3.25z"
      ></path>
    </svg>
  );
};
