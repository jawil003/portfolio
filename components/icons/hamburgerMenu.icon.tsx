import designSystem from "@style/designSystem";
import React from "react";

interface Props {
  width?: string;
  height?: string;
  color?: string;
  className?: string;
}

/**
 * An HamburgerMenuIcon React Component.
 * @author Jannik Will
 * @version 0.1
 */
const HamburgerMenuIcon: React.FC<Props> = ({
  width,
  height,
  color,
  className,
}) => {
  return (
    <svg
      className={`hamburger-menu-icon ${className}`}
      xmlns="http://www.w3.org/2000/svg"
      fillRule="evenodd"
      strokeLinejoin="round"
      strokeMiterlimit="2"
      clipRule="evenodd"
      viewBox="0 0 2481 2481"
      width={width}
      height={height}
      fill={color}
    >
      <path
        d="M966.527 638.508c0-105.526-112.075-191.2-250.12-191.2-138.044 0-250.119 85.674-250.119 191.2V1928.35c0 105.52 112.075 191.2 250.119 191.2 138.045 0 250.12-85.68 250.12-191.2V638.508z"
        transform="matrix(.7503 0 0 .7503 306.782 308.462) matrix(0 -1.1652 1.52426 0 -712.287 2973.76)"
        fill={color}
      ></path>
      <path
        d="M966.527 638.508c0-105.526-112.075-191.2-250.12-191.2-138.044 0-250.119 85.674-250.119 191.2V1928.35c0 105.52 112.075 191.2 250.119 191.2 138.045 0 250.12-85.68 250.12-191.2V638.508z"
        transform="matrix(.7503 0 0 .7503 306.782 308.462) matrix(0 -1.1652 1.52426 0 -712.287 2076.49)"
        fill={color}
      ></path>
      <path
        d="M966.527 638.508c0-105.526-112.075-191.2-250.12-191.2-138.044 0-250.119 85.674-250.119 191.2V1928.35c0 105.52 112.075 191.2 250.119 191.2 138.045 0 250.12-85.68 250.12-191.2V638.508z"
        transform="matrix(.7503 0 0 .7503 306.782 308.462) matrix(0 -1.1652 1.52426 0 -712.287 1179.26)"
        fill={color}
      ></path>
    </svg>
  );
};

HamburgerMenuIcon.defaultProps = {
  color:
    designSystem.colors.brand.secondary,
};

export default HamburgerMenuIcon;
