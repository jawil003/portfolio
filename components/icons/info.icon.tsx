interface Props {
  color?: string;
  size?: string;
}
const InfoIcon: React.FC<Props> = ({
  size,
  color,
}) => {
  return (
    <svg
      width={size}
      height={size}
      viewBox="0 0 426.667 426.667"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M213.333 0C95.467 0 0 95.467 0 213.333s95.467 213.333 213.333 213.333S426.667 331.2 426.667 213.333 331.2 0 213.333 0zm21.334 320H192V192h42.667v128zm0-170.667H192v-42.667h42.667v42.667z"
        fill={color}
        xmlns="http://www.w3.org/2000/svg"
        data-original="#000000"
      />
    </svg>
  );
};

InfoIcon.defaultProps = {
  size: "100%",
  color: "#fff",
};

export default InfoIcon;
