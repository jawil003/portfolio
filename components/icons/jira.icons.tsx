import React from "react";
interface Props {
  fontSize: "large" | "normal" | "small";
}
export const JiraIcon: React.FC<Props> = (
  props: Props
) => {
  return (
    <svg
      style={{
        fontSize:
          props.fontSize === "small"
            ? "0.5rem"
            : props.fontSize === "normal"
            ? "1rem"
            : "3.5rem",
        ...{
          fillRule: "evenodd",
          clipRule: "evenodd",
          strokeLinejoin: "round",
          strokeMiterlimit: 2,
        },
      }}
      width="1em"
      height="1em"
      viewBox="0 0 74 76"
      version="1.1"
    >
      <g id="Layer_2">
        <g id="Blue">
          <path
            d="M72.4,35.76L36.64,0L0.88,35.76C-0.282,36.923 -0.282,38.837 0.88,40L36.64,75.76L72.4,40C73.562,38.837 73.562,36.923 72.4,35.76ZM36.64,49.08L25.44,37.88L36.64,26.68L47.84,37.88L36.64,49.08Z"
            style={{
              fill: "rgb(38,132,255)",
              fillRule: "nonzero",
            }}
          />
          <path
            d="M36.64,26.68C29.355,19.395 29.319,7.419 36.56,0.09L12.05,24.59L25.39,37.93L36.64,26.68Z"
            style={{
              fill: "url(#_Linear1)",
              fillRule: "nonzero",
            }}
          />
          <path
            d="M47.87,37.85L36.64,49.08C40.178,52.616 42.168,57.418 42.168,62.42C42.168,67.422 40.178,72.224 36.64,75.76L61.21,51.19L47.87,37.85Z"
            style={{
              fill: "url(#_Linear2)",
              fillRule: "nonzero",
            }}
          />
        </g>
      </g>
      <defs>
        <linearGradient
          id="_Linear1"
          x1="0"
          y1="0"
          x2="1"
          y2="0"
          gradientUnits="userSpaceOnUse"
          gradientTransform="matrix(-15.64,15.64,-15.64,-15.64,34.64,15.35)"
        >
          <stop
            offset="0"
            style={{
              stopColor: "rgb(0,82,204)",
              stopOpacity: 1,
            }}
          />
          <stop
            offset="0.18"
            style={{
              stopColor: "rgb(0,82,204)",
              stopOpacity: 1,
            }}
          />
          <stop
            offset="1"
            style={{
              stopColor: "rgb(38,132,255)",
              stopOpacity: 1,
            }}
          />
        </linearGradient>
        <linearGradient
          id="_Linear2"
          x1="0"
          y1="0"
          x2="1"
          y2="0"
          gradientUnits="userSpaceOnUse"
          gradientTransform="matrix(383.538,-591.775,591.775,383.538,989.465,2323.06)"
        >
          <stop
            offset="0"
            style={{
              stopColor: "rgb(0,82,204)",
              stopOpacity: 1,
            }}
          />
          <stop
            offset="0.18"
            style={{
              stopColor: "rgb(0,82,204)",
              stopOpacity: 1,
            }}
          />
          <stop
            offset="1"
            style={{
              stopColor: "rgb(38,132,255)",
              stopOpacity: 1,
            }}
          />
        </linearGradient>
      </defs>
    </svg>
  );
};
