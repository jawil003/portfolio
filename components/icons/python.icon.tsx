interface Props {
  fontSize: "large" | "normal" | "small";
}
export const PythonLogo: React.FC<Props> = ({
  fontSize,
}) => {
  return (
    <svg
      style={{
        fontSize:
          fontSize === "small"
            ? "0.5rem"
            : fontSize === "normal"
            ? "1rem"
            : "3.5rem",
      }}
      width="1em"
      height="1em"
      xmlns="http://www.w3.org/2000/svg"
      fillRule="evenodd"
      strokeLinejoin="round"
      strokeMiterlimit="2"
      clipRule="evenodd"
      viewBox="0 0 134 126"
    >
      <g fillRule="nonzero" transform="translate(6 1)">
        <path
          fill="url(#_Linear1)"
          d="M60.51 6.398c-4.583.021-8.961.412-12.812 1.094-11.347 2.004-13.407 6.2-13.407 13.937v10.219h26.813v3.406H24.229c-7.793 0-14.616 4.684-16.75 13.594-2.462 10.213-2.571 16.586 0 27.25 1.906 7.938 6.457 13.594 14.25 13.594h9.219v-12.25c0-8.85 7.657-16.657 16.75-16.657h26.781c7.455 0 13.406-6.138 13.406-13.625V21.429c0-7.266-6.13-12.725-13.406-13.937-4.606-.767-9.385-1.115-13.969-1.094zm-14.5 8.219c2.77 0 5.031 2.298 5.031 5.125 0 2.816-2.261 5.093-5.031 5.093-2.779 0-5.031-2.277-5.031-5.093 0-2.827 2.252-5.125 5.031-5.125z"
        ></path>
        <path
          fill="url(#_Linear2)"
          d="M91.229 35.054V46.96c0 9.231-7.826 17-16.75 17H47.698c-7.336 0-13.407 6.279-13.407 13.625v25.532c0 7.266 6.319 11.54 13.407 13.625 8.487 2.495 16.626 2.946 26.781 0 6.75-1.955 13.406-5.888 13.406-13.625V92.898H61.104v-3.406h40.187c7.793 0 10.697-5.436 13.407-13.594 2.799-8.399 2.68-16.476 0-27.25-1.926-7.757-5.604-13.594-13.407-13.594H91.229zM76.166 99.71c2.78 0 5.032 2.278 5.032 5.094 0 2.827-2.252 5.125-5.032 5.125-2.769 0-5.031-2.298-5.031-5.125 0-2.816 2.262-5.094 5.031-5.094z"
        ></path>
      </g>
      <defs>
        <linearGradient
          id="_Linear1"
          x1="0"
          x2="1"
          y1="0"
          y2="0"
          gradientTransform="scale(81.2332) rotate(40.98 -.071 .131)"
          gradientUnits="userSpaceOnUse"
        >
          <stop offset="0" stopColor="#5A9FD4"></stop>
          <stop offset="1" stopColor="#306998"></stop>
        </linearGradient>
        <linearGradient
          id="_Linear2"
          x1="0"
          x2="1"
          y1="0"
          y2="0"
          gradientTransform="scale(-38.1867) rotate(55.006 1.625 -3.26)"
          gradientUnits="userSpaceOnUse"
        >
          <stop offset="0" stopColor="#FFD43B"></stop>
          <stop offset="1" stopColor="#FFE873"></stop>
        </linearGradient>
      </defs>
    </svg>
  );
};
