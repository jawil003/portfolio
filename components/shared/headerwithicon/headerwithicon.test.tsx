import { render } from "@testing-library/react";
import { AcUnit } from "@material-ui/icons";
import {
  HeaderWithIcon,
  HeaderWithIconProps,
} from ".";

jest.mock(
  "react-intersection-observer",
  () => ({
    useInView: jest
      .fn()
      .mockResolvedValue({ ref: {} }),
  }),
);

const variants: HeaderWithIconProps[] = [
  {
    first: true,
    icon: <AcUnit />,
    arrow: true,
  },
  {
    first: false,
    icon: <AcUnit />,
    arrow: true,
  },
  {
    first: true,
    icon: <AcUnit />,
    arrow: false,
  },
];

describe("HeaderWithIcon", () => {
  test.each(variants)(
    "Create HeaderWithIcon with %p",
    (variant) => {
      const testheaderwithicon = render(
        <HeaderWithIcon {...variant} />,
      );

      expect(
        testheaderwithicon.container
          .firstChild,
      ).toMatchSnapshot();
    },
  );
});
