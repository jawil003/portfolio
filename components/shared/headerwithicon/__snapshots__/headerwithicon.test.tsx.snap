// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderWithIcon Create HeaderWithIcon with {"arrow": false, "first": true, "icon": [Object]} 1`] = `
.emotion-0 {
  margin-top: 100px;
  height: calc(
                  100% - 100px
                );
  width: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

@media (min-width: 600px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .emotion-0>* {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-0>:first-of-type {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

@media (max-width: 900px) {
  .emotion-0>:last-child {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    z-index: -1;
    opacity: 0.3;
    padding: 0px 25%;
    height: 100%;
    overflow: hidden;
  }

  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .emotion-0>* {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-0 h1,
  .emotion-0 h2,
  .emotion-0 h3.h4,
  .emotion-0 h5,
  .emotion-0 h6,
  .emotion-0 p,
  .emotion-0 span {
    text-align: center;
  }

  .emotion-0>:first-of-type {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 100px;
}

<header
  class="emotion-0"
>
  <div
    style="opacity: 0; transform: translateY(-200px) translateZ(0);"
  />
  <div
    class="emotion-1"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
      />
    </svg>
  </div>
</header>
`;

exports[`HeaderWithIcon Create HeaderWithIcon with {"arrow": true, "first": false, "icon": [Object]} 1`] = `
.emotion-0 {
  margin-top: 100px;
  height: calc(
                  100% - 100px
                );
  width: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

@media (min-width: 600px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .emotion-0>* {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-0>:first-of-type {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

@media (max-width: 900px) {
  .emotion-0>:last-child {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    z-index: -1;
    opacity: 0.3;
    padding: 0px 25%;
    height: 100%;
    overflow: hidden;
  }

  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .emotion-0>* {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-0 h1,
  .emotion-0 h2,
  .emotion-0 h3.h4,
  .emotion-0 h5,
  .emotion-0 h6,
  .emotion-0 p,
  .emotion-0 span {
    text-align: center;
  }

  .emotion-0>:first-of-type {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 100px;
}

<header
  class="emotion-0"
>
  <div
    style="opacity: 0; transform: translateY(-200px) translateZ(0);"
  />
  <div
    class="emotion-1"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
      />
    </svg>
  </div>
</header>
`;

exports[`HeaderWithIcon Create HeaderWithIcon with {"arrow": true, "first": true, "icon": [Object]} 1`] = `
.emotion-0 {
  margin-top: 100px;
  height: calc(
                  100% - 100px
                );
  width: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

@media (min-width: 600px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .emotion-0>* {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-0>:first-of-type {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

@media (max-width: 900px) {
  .emotion-0>:last-child {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    z-index: -1;
    opacity: 0.3;
    padding: 0px 25%;
    height: 100%;
    overflow: hidden;
  }

  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .emotion-0>* {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-0 h1,
  .emotion-0 h2,
  .emotion-0 h3.h4,
  .emotion-0 h5,
  .emotion-0 h6,
  .emotion-0 p,
  .emotion-0 span {
    text-align: center;
  }

  .emotion-0>:first-of-type {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 100px;
}

<header
  class="emotion-0"
>
  <div
    style="opacity: 0; transform: translateY(-200px) translateZ(0);"
  />
  <div
    class="emotion-1"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
      />
    </svg>
  </div>
</header>
`;
